src/metrics/chaoss.ts

```diff
@@ -411,4 +411,54 @@ FROM
         lines,
       }
     });
-  };
\ No newline at end of file
+  };
+
+  export const chaossNewContributors = async (config: QueryConfig) => {
+    config = getMergedConfig(config);
+    const whereClauses: string[] = ["type = 'PushEvent'"];
+    const repoWhereClause = getRepoWhereClauseForClickhouse(config);
+    if (repoWhereClause) whereClauses.push(repoWhereClause);
+    // whereClauses.push(getTimeRangeWhereClauseForClickhouse(config));
+    const endDate = new Date(`${config.endYear}-${config.endMonth}-1`);
+    endDate.setMonth(config.endMonth);  // find next month
+    
+    const sql = `
+    SELECT
+      id,
+      argMax(name, time) AS name,
+      SUM(count) AS total_count,
+      ${getGroupArrayInsertAtClauseForClickhouse(config, { key: 'new_contributors', value: 'count' })}
+    FROM
+    (
+      SELECT
+        ${getGroupTimeAndIdClauseForClickhouse(config, 'repo')},
+        count() as count
+        FROM(
+         SELECT min(created_at)as created_at,org_id,repo_id,name
+         FROM(
+          SELECT created_at,org_id,repo_id,arrayJoin(push_commits.name) as name
+          FROM github_log.events
+          WHERE ${whereClauses.join(' AND ')}
+           )AS t
+         GROUP BY org_id,repo_id,name
+      )
```

Comment:
- Please make sure that any SQL keyword should be in uppercase, like `COUNT`, `MIN`, `AS`.
- Please make sure to leave a blank space between columns, like `MIN(created_at) AS created_at, org_id, repo_id`, and etc.
- Actually I think `MIN(created_at)` and `arrayJoin` can be in a single SQL like:
```sql
SELECT MIN(created_at) AS created_at, org_id, repo_id, arrayJoin(push_commits.name) AS name
...
GROUP BY org_id, repo_id, name
```